test/e2e-tests.sh

```diff
@@ -81,7 +81,7 @@ kubectl create namespace serving-tests
 options=""
 (( EMIT_METRICS )) && options="-emitmetrics"
 report_go_test \
-  -v -tags=e2e -count=1 -timeout=20m \
+  -v -tags=e2e -count=1 -timeout=40m \
```

Comment:
We need to split these tests. 30+ minutes for the whole test suite is starting to hurt productivity.

Reply:
The controller reboot tests cannot run in parallel with other tests.  But most of the other tests can.  Should we separate the longer tests from the short ones?

Alternatively, we can implement controllers for each namespace: https://github.com/knative/serving/issues/2107 and tests in different namespaces.

Reply:
@evankanderson, @mattmoor, @vaikas-google what do you think about running separate controllers in each namespace (configurable)?

Reply:
That is long lead (longer than we should wait).

@adrcunha what is your recommendation for splitting things up?

Reply:
> The controller reboot tests cannot run in parallel with other tests. But most of the other tests can. Should we separate the longer tests from the short ones?

Don't the conformance and e2e tests already run in parallel on the same cluster? Or am I misremembering the current test setup?

Reply:
I think you're right @bbrowning, the "DiagnoseMe" output actually proves that there are occasionally tests running in parallel to the AutoscaleUpDown test.

Reply:
@adrcunha, what should we do here? Split out the reboot tests that have to run serially?

Reply:
Quickest fix is to run the reboot tests after the other e2e/conformance tests. But I feel that we actually need to split our e2e test job to avoid the 30+ run time.

Reply:
I added the required infra to run these tests separate from the e2e job, and the upgrade tests are already using it.